### DISCUSSION TITLE
Reproducing mug.e mesh file in 'Example 1: As Simple as it Gets' with Gmsh

### DISCUSSION THREAD

**Discussion by kurtenkera** [idaholab/moose#26950] [Q&A Meshing]
I'm trying to reproduce 'Example 1: As Simple as it Gets' (https://mooseframework.inl.gov/getting_started/examples_and_tutorials/examples/ex01_inputfile.html), but by creating the mug.e mesh file parametrically (i.e., without a GUI) with Gmsh in a Python script. I'm doing this because I plan to use Gmsh to mesh up more complex geometries parametrically when using MOOSE in the future, so I want to test my use of Gmsh + MOOSE on this simple example. But as I have never actually generated a finite element mesh before, I have a few questions:

1) How can I view the mug.e mesh in the format seen below? I have tried simply reading the mug.e file into Paraview but couldn't reproduce the same figure below (which clearly shows the meshed cylinder whose elements are cube-shaped)?

![image](https://github.com/idaholab/moose/assets/127472392/2d48e0a9-743d-49ef-9026-e46efe1f03a5)

2) Will it be crucial that I create a cylinder with the same number of elements and also the same cube-shaped elements as seen above if I want to reproduce the same results? I ask this because currently I have created a cylinder (yet to be hollowed out) in Gmsh with tetrahedron-shaped elements (see below), and I feel like even once I hollow out the cylinder, the results mightn't be identical because the elements have different shapes?

![image](https://github.com/idaholab/moose/assets/127472392/b89c0183-1545-4206-b972-5879f38ac32f)

3) Where can I find out how specifically the mug.e file was created (i.e. it should have been created using CUBIT from Sandia National Laboratories if it is a .e file no?) and any associated code if the mug.e file was produced with a `parametric meshing' approach?

Thanks in advance!

**Comment by GiudGiud** [comment]
1 paraview should be fine to visualize software. It has plenty of options, you can do a lot of things with it

2 within the limit of fine refinement you should get the same results between tets and hexes.

**Reply by GiudGiud** [reply]
I don't know who generated the mug mesh. This is probably a pretty generic problem so i would check the manual and the examples of cubit to see if it s not included there
